src/main/java/com/joss/conductor/mobile/annotation/PlatformIdHelper.java

```diff
@@ -0,0 +1,53 @@
+package com.joss.conductor.mobile.annotation;
+
+import com.joss.conductor.mobile.Platform;
+import org.openqa.selenium.By;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+/**
+ * Created on 3/14/18.
+ */
+public final class PlatformIdHelper {
+
+    public static void initIds(Object page, Platform platform) throws Exception {
+
+        Class current = page.getClass();
+        do {
```

Comment:
Is there a reason to use a `do` loop instead of an `if` statement? It seems to me that we'll try to access `current` before `null` checking.

Reply:
Nvm, just read the final line before the closing curly brace, maybe still check and make sure `current` is not null before trying to access it. Although on initial call it _should_ never be `null`...I am torn.

Reply:
I suppose if it does end up being `null` it'll fail either way so it might be good to handle it gracefully if it isn't already.

Reply:
todo

Reply:
@jossjacobo It's impossible for it to be null. If `page` is null then a NullPointerException is thrown.